test/dynamo/test_repros.py

```diff
@@ -3594,6 +3594,62 @@ def fn(x=None):
         self.assertEqual(opt_fn("10"), fn("10"))
         self.assertEqual(cnt.frame_count, 4)
 
+    def test_tensor_set_data(self):
+        # https://github.com/pytorch/pytorch/issues/113030
```

Comment:
Could you also add an `x += 1` before `x.data`? This will currently create a bug 

https://github.com/pytorch/pytorch/issues/113160

Reply:
Sure, but let's solve the other things in this PR first :) we can always stack more on top.